client/js/lounge.js

```diff
@@ -465,7 +471,35 @@ $(function() {
 		return false;
 	});
 
+	sidebar.on("click", ".add-channel", (e) => {
+		const id = $(e.target).data("id");
+		const joinForm = $(`#join-channel-${id}`);
+		joinForm.toggle();
+		joinForm.find(".input[name='channel']").focus();
+		return false;
+	});
+
+	sidebar.on("submit", ".join-form", function() {
+		const form = $(this);
+		const channel = form.find("input[name='channel']").val();
+		const key = form.find("input[name='key']").val();
+		const chan = utils.findCurrentNetworkChan(channel);
+		if (chan.length) {
+			chan.click();
+		} else {
+			socket.emit("input", {
+				text: `/join ${channel} ${key}`,
+				target: form.prev().data("id"),
+			});
+		}
+		form.hide();
```

Comment:
Could you clear the fields after submitting? It currently keeps the former values.